src/commands/emoji.ts

```diff
@@ -0,0 +1,178 @@
+  @Inject private emojiDatabaseService: EmojiDatabaseService;
+
+  async execute(cmdArgs: ICommandArgs): Promise<ICommandResult> {
+    const { message } = cmdArgs;
+    const args = cmdArgs.args.split(' ');
+
+    // TODO: Fix this args check
+    let queryType = args[0];
+    let result = [];
+    let time = 0;
+
+    // Check for time
+    if (args.length && this.hasTime(args)) {
+        let timeVal = args.pop();
+        queryType = queryType.replace(timeVal, '');
+        time = this.convertToSeconds(timeVal);
+    }
+
+    const isReaction = ['r', 're', 'ru'].includes(queryType) || queryType.includes('reaction');
+
+    if (args.length === 1 && ['user', 'u', 'reactionuser', 'reactionsuser', 'ru', 'r', 'reaction', 'reactions'].includes(queryType)) {
+        args.push(message.author.id);
+    }
+
+    if (args.length !== 2 && !['s', 'server', ''].includes(queryType)) {
+        // incorrect format
+        this.logger.log('Incorrect format');
+        return;
+    }
+
+    switch (queryType) {
+        case 'user': case 'u': case 'ru': case 'reactionuser': case 'reactionsuser': {
+            // TODO: Validate username input [args[1] into username]
+            this.logger.log('User lookup');
+            result = await this.emojiDatabaseService.userLookup(args[1], message.guild.id, time, isReaction);
+            // this.logger.log(await this.emojiDatabaseService.userLookup(args[1], message.guild.id, isReaction));
+            break;
+        }
+
+        case 'emoji': case 'e': case 're': case 'reactionemoji': case 'reactionsemoji': {
+            this.logger.log('Emoji lookup');
+            result = await this.emojiDatabaseService.emojiLookup(args[1], message.guild.id, time, isReaction);
+            // this.logger.log(await this.emojiDatabaseService.emojiLookup(args[1], message.guild.id, isReaction));
+            break;
+        }
+
+        case 'server': case 's': case '': case 'reactions': case 'reaction': case 'r': {
+            this.logger.log('Server lookup');
+            result = await this.emojiDatabaseService.serverLookup(message.guild.id, time, isReaction);
+            // this.logger.log(await this.emojiDatabaseService.serverLookup(message.guild.id, isReaction));
+            break;
+        }
+
+        default: {
+            this.logger.log('Invalid option');
+            // Invalid option
+            break;
+        }
+    }
+
+    let formatResult = await this.formatEmojiData(result, queryType);
+
+    if (formatResult.length) {
+        this.paginateEmojiData(message, formatResult);
+    } else {
+        message.channel.send('No data found for specified query');
+    }
+
+    return {};
+  }
+
+  private paginateEmojiData(message, formattedResults) {
+    const embed = Pagination(message, formattedResults);
+  }
+
+  private async formatEmojiData(emoji, queryType) {
+    // TODO: use queryType to determine what to display
+    let valueList = [];
+    let value;
+
+    while (emoji.length) {
```

Comment:
this looks really complicated, not sure why this while loop is here if the inner loop is a big switch statement (essentially)